randovania/server/user_session.py

```diff
@@ -329,18 +338,53 @@ async def browser_discord_login_callback(
     return unable_to_login(sa, request, error_message, status_code)
 
 
-@router.get("/me", response_class=HTMLResponse)
-async def browser_me(sa: ServerAppDep, request: Request, user: UserDep) -> HTMLResponse:
+@router.get("/me")
+async def browser_me(sa: ServerAppDep, request: Request, user: UserDep) -> Response:
+    if sa.is_api_request(request):
+        return JSONResponse(user.as_json)
+
     return sa.templates.TemplateResponse(
         request,
         "user_session/me.html.jinja",
         context={
             "user": user,
-            "is_admin": user.admin or sa.app.debug,
+            "is_admin": user.admin,
         },
     )
 
 
+@router.get("/guest_login")
+async def guest_login(sa: ServerAppDep, request: Request) -> Response:
+    if not sa.app.debug:
```

Comment:
this is inconsistent with the check performed by the existing login method: https://github.com/randovania/randovania/blob/76f51f214ed5dd9f34131f18bab878b48005dcb6/randovania/server/user_session.py#L119-L121